Objects/dictobject.c

```diff
@@ -1597,7 +1597,7 @@ insertion_resize(PyInterpreterState *interp, PyDictObject *mp, int unicode)
 }
 
 static Py_ssize_t
-insert_into_splitdictkeys(PyDictKeysObject *keys, PyObject *name)
+insert_into_splitdictkeys_locked(PyDictKeysObject *keys, PyObject *name)
```

Comment:
I think this may be cleaner if you take a `Py_hash_t hash` as an argument here and do the hash computation in `_PyObject_StoreInstanceAttribute` in all builds instead of in only the GIL-disabled build.